seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/catalog/HiveJDBCCatalog.java

```diff
@@ -0,0 +1,403 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.hive.catalog;
+
+import org.apache.seatunnel.api.configuration.ReadonlyConfig;
+import org.apache.seatunnel.api.table.catalog.Catalog;
+import org.apache.seatunnel.api.table.catalog.CatalogTable;
+import org.apache.seatunnel.api.table.catalog.PhysicalColumn;
+import org.apache.seatunnel.api.table.catalog.TableIdentifier;
+import org.apache.seatunnel.api.table.catalog.TablePath;
+import org.apache.seatunnel.api.table.catalog.TableSchema;
+import org.apache.seatunnel.api.table.catalog.exception.CatalogException;
+import org.apache.seatunnel.api.table.catalog.exception.DatabaseAlreadyExistException;
+import org.apache.seatunnel.api.table.catalog.exception.DatabaseNotExistException;
+import org.apache.seatunnel.api.table.catalog.exception.TableAlreadyExistException;
+import org.apache.seatunnel.api.table.catalog.exception.TableNotExistException;
+import org.apache.seatunnel.connectors.seatunnel.hive.sink.HiveSinkOptions;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.Serializable;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HiveJDBCCatalog implements Catalog, Serializable {
+
+    private final String catalogName = "Hive";
+    private final ReadonlyConfig config;
+
+    private Connection connection;
+
+    public HiveJDBCCatalog(ReadonlyConfig config) {
+        try {
+            Class.forName("org.apache.hive.jdbc.HiveDriver");
+        } catch (ClassNotFoundException e) {
+            throw new CatalogException(e);
+        }
+        this.config = config;
+    }
+
+    public HiveTable getTableInformation(TablePath tablePath, boolean throwExceptionIfNotExist) {
+        if (!tableExists(tablePath)) {
+            if (throwExceptionIfNotExist) {
+                throw new TableNotExistException(catalogName, tablePath);
+            } else {
+                return null;
+            }
+        }
+        TableSchema.Builder builder = TableSchema.builder();
+        List<String> partitionKeys = new ArrayList<>();
+        String describeFormattedTableQuery = "describe formatted " + tablePath.getFullName();
+        try (PreparedStatement ps = connection.prepareStatement(describeFormattedTableQuery)) {
+            ResultSet rs = ps.executeQuery();
+            return processResult(rs, tablePath, builder, partitionKeys);
```

Comment:
Now hive table informaction is parse from the query result. That's not very elegant, but it work.....